src/main/game/Board.java

```diff
@@ -46,96 +57,84 @@ public boolean hasAvailableMoves() {
         int count = 0;
 
         for (String space : this.grid) {
-            if (space.equals(PLAYER_ONE_MARK.getPlayer()) || space.equals(PLAYER_TWO_MARK.getPlayer())) {
+            if (space.equals(X_MARK.getPlayer()) || space.equals(O_MARK.getPlayer())) {
                 count += 1;
             }
         }
         return count != grid.size();
     }
 
-    public boolean horizontalWin(String playerMark) {
-        List<List<String>> rows = getRows();
-
-        return findWin(playerMark, rows);
-
-    }
-
-    public List<List<String>> getColumns() {
-        List<List<String>> rows = new ArrayList<>();
+    public List<Line> getColumnLines() {
+        List<Line> columns = new ArrayList<>();
 
         for (int i = 0; i < ROW_COUNT; i++) {
-            List<String> row = asList(grid.get(i), grid.get(i + 3), grid.get(i + 6));
-            rows.add(row);
+            Line line = new Line(grid.get(i), grid.get(i + 3), grid.get(i + 6));
+            columns.add(line);
         }
-        return rows;
+        return columns;
     }
 
-    public boolean columnWin(String playerMark) {
-        List<List<String>> columns = getColumns();
-
-        return findWin(playerMark, columns);
-    }
-
-    private boolean findWin(String playerMark, List<List<String>> rows) {
-        for (List<String> line : rows) {
-            if (win(playerMark, line)) {
+    public boolean findWin(String playerMark) {
+        for (Line line : lines()) {
+            if (line.hasWinner(playerMark)) {
                 return true;
             }
         }
         return false;
     }
 
-    public List<List<String>> getDiagonals() {
-        List<List<String>> diagonals = new ArrayList<>();
-        List<String> diagonalOne = asList(grid.get(0), grid.get(4), grid.get(8));
-        List<String> diagonalTwo = asList(grid.get(2), grid.get(4), grid.get(6));
+    public List<Line> getDiagonalsLines() {
+        List<Line> diagonals = new ArrayList<>();
+        Line diagonalOne = new Line(grid.get(0), grid.get(4), grid.get(8));
+        Line diagonalTwo = new Line(grid.get(2), grid.get(4), grid.get(6));
         diagonals.add(diagonalOne);
         diagonals.add(diagonalTwo);
         return diagonals;
     }
 
-    public boolean diagonalWin(String playerMark) {
-        List<List<String>> diagonals = getDiagonals();
-
-        return findWin(playerMark, diagonals);
-
+    public List<String> getAvailableMoves() {
+        List<String> availableMoves = new ArrayList<>();
+        for (String space : this.grid) {
+            if (!space.equals(X_MARK.getPlayer()) && !space.equals(O_MARK.getPlayer())) {
```

Comment:
Similar comment to the conditional above `cellTaken` which you could reuse here if you extracted from the above.

Reply:
I wonder if you could go even further.  What you are actually checking here is that the space/cell is not taken by either mark, which could be taken further to say that you are actually testing whether the space is empty, so it would be good if the expression read as such.

Though I think this pulls up a naming issue.  Are X and O really player types?  If they were called something else, it might make sense to have X, O and something that represented neither?